homeassistant/components/myuplink/select.py

```diff
@@ -0,0 +1,98 @@
+"""Select entity for myUplink."""
+
+from aiohttp import ClientError
+from myuplink import DevicePoint
+
+from homeassistant.components.select import SelectEntity, SelectEntityDescription
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import MyUplinkConfigEntry, MyUplinkDataCoordinator
+from .entity import MyUplinkEntity
+from .helpers import find_matching_platform, skip_entity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: MyUplinkConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up myUplink select."""
+    entities: list[SelectEntity] = []
+    coordinator: MyUplinkDataCoordinator = config_entry.runtime_data
+
+    # Setup device point select entities
+    for device_id, point_data in coordinator.data.points.items():
+        for point_id, device_point in point_data.items():
+            if skip_entity(device_point.category, device_point):
+                continue
+            description = None
+            if find_matching_platform(device_point, description) == Platform.SELECT:
+                entities.append(
+                    MyUplinkSelect(
+                        coordinator=coordinator,
+                        device_id=device_id,
+                        device_point=device_point,
+                        entity_description=description,
+                        unique_id_suffix=point_id,
+                    )
+                )
+
+    async_add_entities(entities)
+
+
+class MyUplinkSelect(MyUplinkEntity, SelectEntity):
+    """Representation of a myUplink select entity."""
+
+    def __init__(
+        self,
+        coordinator: MyUplinkDataCoordinator,
+        device_id: str,
+        device_point: DevicePoint,
+        entity_description: SelectEntityDescription | None,
+        unique_id_suffix: str,
+    ) -> None:
+        """Initialize the select."""
+        super().__init__(
+            coordinator=coordinator,
+            device_id=device_id,
+            unique_id_suffix=unique_id_suffix,
+        )
+
+        # Internal properties
+        self.point_id = device_point.parameter_id
+        self._attr_name = device_point.parameter_name
+
+        self._attr_options = [x["text"].capitalize() for x in device_point.enum_values]
```

Comment:
I wonder if you should lowercase instead, to make it future translatable?

Reply:
Actually, we should make this translatable from the get-go.

Reply:
This topic has been discussed previously for this integration. Entity names, enum sensors and option values are created from the metadata in the API. However there are huge numbers (number of supported heat pump models * (20 to 600 entities) of possible entity names in the API, they are normally easy to understand for a user that is interested in using them. E.g. "Discharge (EB101-BT14)". There is no official documentation of the data points (=entities) in the API and you need access to each and every appliance model in your account to harvest all possible values. 
Thus, the decision up to now has been just stay with the english terms and not enable translations. I would rather continue this way in this PR, which by the way probably is that last platform needed to get at complete user experience. If we should enable translations fully in this integration it should be done in a separate PR or series of PR:s.

Reply:
> Thus, the decision up to now has been just stay with the english terms and not enable translations. 

Let's document that in code using comments, so we don't have to discuss this in reviews in the future again 😉

Reply:
Since this comment will be applicable to all platform modules I think it is better to make it in a separate PR.